### PR TITLE
Remove task arguments "one-liner" syntax

### LINKED ISSUE #492: Remove task arguments "one-liner" syntax
Now that we introduced the new DSL changes, the one-liner task remains inelegant (we always need to use the `${ expr }` syntax for all the arguments...)
So we decided to remove that option from the language to not cause confusion

### PR DESCRIPTION
**Pull request by maayan-avraham**
Fixes #492 
Task arguments can be defined only using a standard YAML list.
Signed-off-by: Maayan Avraham maayan.avraham@hpe.com

### CODE REVIEW

**Comment by maayan-avraham** [comment]
@Bonczidai please review

**Comment by orius123** [comment]
Y?

**Comment by oritstone** [comment]
We decided to remove this syntax since the syntax was not that elegant to begin with, and now with the changes to the expression and constant values syntax, it forced us to have task one liner arguments to always have an expression syntax:
```yaml
do:
  flow: input1 = ${"hi"}, input2 =  ${3}, input3 = ${expression}
```
as opposed to the syntax of the list of args, which is confusing.
Since this syntax was anyway not in use yet, we decided to remove it

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DoTransformer.java [hunk 1]
```diff
@@ -16,7 +16,6 @@
 
 import io.cloudslang.lang.entities.bindings.Argument;
 import org.apache.commons.collections4.MapUtils;
-import org.apache.commons.lang.StringUtils;
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DoTransformer.java]: you can remove `REGEX_COMMA_NOT_ESCAPED` unused const

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DoTransformer.java [hunk 2]
```diff
@@ -47,8 +46,7 @@
                 result.add(argument);
             }
         } else if (rawArguments instanceof String) {
-            // one liner syntax
-            result = transformOneLinerArguments((String) rawArguments);
+            throw new RuntimeException("Task arguments should be defined using a standard YAML list.");
```
Comment (Bonczidai) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/transformers/DoTransformer.java]: throw an error when the value is not a list and not null - how about this:
```
        } else if (rawArguments != null) {
            throw new RuntimeException("Task arguments should be defined using a standard YAML list.");
        }
```

so we will allow cases:
```
    - CheckWeather:
        do:
          ops.test_op:
```
and
```
    - CheckWeather:
        do:
          ops.test_op: [....]
```

File: cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_contains_reserved_characters.yaml [hunk 3]
```diff
@@ -11,14 +11,16 @@ imports:
   ops: user.ops
 
 flow:
-  name: flow_missing_argument_name
+  name: flow_arguments_contains_reserved_characters
```
Comment (Bonczidai) [inline cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_contains_reserved_characters.yaml]: what are those reserved chars? can we remove this flow?

File: cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_not_in_list.yaml [hunk 4]
```diff
@@ -11,7 +11,7 @@ imports:
   ops: user.ops
 
 flow:
-  name: flow_1
+  name: flow_arguments_not_in_list
```
Comment (Bonczidai) [inline cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_not_in_list.yaml]: remove? contains:
```
    - CheckWeather:
        do:
          ops.check_Weather: city = ${ city_name }, country = ${ 'Wonderland' }
        publish:
          - weather
```

File: cloudslang-tests/src/test/java/io/cloudslang/lang/systemtests/SimpleFlowTest.java [hunk 5]
```diff
@@ -114,10 +114,10 @@ private void compileAndRunSimpleFlowOneLinerSyntax(Map<String, ? extends Seriali
         URI operations1 = getClass().getResource("/yaml/get_time_zone.sl").toURI();
         URI operations2 = getClass().getResource("/yaml/comopute_daylight_time_zone.sl").toURI();
```
Comment (Bonczidai) [inline cloudslang-tests/src/test/java/io/cloudslang/lang/systemtests/SimpleFlowTest.java]: typo here: comopute

File: cloudslang-tests/src/test/java/io/cloudslang/lang/systemtests/SimpleFlowTest.java [hunk 6]
```diff
@@ -50,7 +50,7 @@ public void testSimpleFlowBasic() throws Exception {
     }
 
     @Test(timeout = DEFAULT_TIMEOUT)
-    public void testSimpleFlowBasicOneLinerSyntax() throws Exception {
+    public void testSimpleFlowInvalidOneLinerSyntax() throws Exception {
```
Comment (Bonczidai) [inline cloudslang-tests/src/test/java/io/cloudslang/lang/systemtests/SimpleFlowTest.java]: Are you trying to test here that one-liner syntax is no more valid? (we should have a test for that)
maybe change the name to `testOneLinerIsInvalid`? WDYT?

**Comment by Bonczidai** [comment]
can we remove `cloud-slang\cloudslang-tests\src\test\resources\yaml\simple_flow_one_liner_binding_check.yaml` and `input_concat_2_one_liner` from `cloud-slang\cloudslang-tests\src\test\resources\yaml\formats\values_op.sl` ?

**Comment by orius123** [comment]
Actually this 'un elegant' syntax is what ansible uses. And if you have string inputs you dont need the expression marker.

File: cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_not_in_list.yaml (hunk 4 above)
Comment (maayan-avraham) [inline cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_not_in_list.yaml]: no. it's for testing that one liner transform is not valid. I changed the flow name to be more clear.

File: cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_contains_reserved_characters.yaml (hunk 3 above)
Comment (maayan-avraham) [inline cloudslang-compiler/src/test/resources/task-args-in-list/flow_arguments_contains_reserved_characters.yaml]: I agree -- there are no reserved characters in the arguments. I'm removing it

**Comment by maayan-avraham** [comment]
@Bonczidai please review again and merge.

**Comment by Bonczidai** [comment]
:+1:

**Pull request merged**